lambda/models/state_machine/create_model.py

```diff
@@ -97,14 +100,69 @@ def handle_poll_docker_image_available(event: Dict[str, Any], context: Any) -> D
 def handle_start_create_stack(event: Dict[str, Any], context: Any) -> Dict[str, Any]:
     """Start model infrastructure creation."""
     output_dict = deepcopy(event)
+    request = CreateModelRequest.validate(event)
+
+    prepared_event = {}
+
+    def camelize_object(o):
+        o2 = {}
+        for k in o:
+            fixed_k = k[:1].lower() + k[1:]
+            if isinstance(o[k], dict):
+                o2[fixed_k] = camelize_object(o[k])
+            else:
+                o2[fixed_k] = o[k]
+        return o2
+
+    prepared_event = camelize_object(event)
+    prepared_event["containerConfig"]["environment"] = event["ContainerConfig"]["Environment"]
+    prepared_event["containerConfig"]["image"] = {
+        "repositoryArn": os.environ["ECR_REPOSITORY_ARN"],
+        "tag": event["image_info"]["image_tag"],
+        "type": "ecr",
+    }
+
+    response = lambdaClient.invoke(
+        FunctionName=os.environ["ECS_MODEL_DEPLOYER_FN_ARN"],
+        Payload=json.dumps({"modelConfig": prepared_event}),
+    )
+
+    payload = response["Payload"].read()
+    payload = json.loads(payload)
+    stack_name = payload.get("stackName")
+
+    response = cfnClient.describe_stacks(StackName=stack_name)
+    stack_arn = response["Stacks"][0]["StackId"]
+    output_dict["stack_name"] = stack_name
+    output_dict["stack_arn"] = stack_arn
+
+    model_table.update_item(
+        Key={"model_id": request.ModelId},
+        UpdateExpression="SET StackName = :stack_name, StackArn = :stack_arn",
+        ExpressionAttributeValues={":stack_name": {"S": stack_name}, ":stack_arn": {"S": stack_arn}},
+    )
+
+    output_dict["remaining_polls_stack"] = 30
+
     return output_dict
 
 
 def handle_poll_create_stack(event: Dict[str, Any], context: Any) -> Dict[str, Any]:
     """Check that model infrastructure creation has completed or not."""
     output_dict = deepcopy(event)
-    output_dict["continue_polling_stack"] = False
-    return output_dict
+    response = cfnClient.describe_stacks(StackName=event["stack_name"])
+    stackStatus = response["Stacks"][0]["StackStatus"]
+    if stackStatus in ["CREATE_COMPLETE", "UPDATE_COMPLETE"]:
+        output_dict["continue_polling_stack"] = False
+        return output_dict
+    elif stackStatus in ["CREATE_IN_PROGRESS", "UPDATE_IN_PROGRESS"]:
+        output_dict["continue_polling_stack"] = True
+        output_dict["remaining_polls_stack"] -= 1
+        if output_dict["remaining_polls_stack"] <= 0:
+            raise Exception("Maximum number of CloudFormation polls reached")
+        return output_dict
+    else:
+        raise Exception("Stack in unexpected state.")
```

Comment:
Can we put the stack status in the error message too? Just to make it a little easier to see what state we're in? I think we may find some other states, but for now, I think these are ok to start- I would just like the status logged if there _is_ something awry

Reply:
good call, added